### PR TITLE
Group feature IDs in 256-bit unit

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This PR changes to make groups of feature IDs in 256-bit units so that `vmovdqa` is more likely to be called instead of `vmovdqu`.

Benchmark results:
* main: 7.764 [sec]
* this branch: 7.035 [sec]

### CODE REVIEW

**Comment by vbkaisetsu** [comment]
I renamed `U31Array` to `U31x8` and added a new struct named `U31`.

File: vibrato/src/dictionary/connector/raw_connector/scorer.rs [hunk 1]
```diff
@@ -3,13 +3,112 @@ use std::collections::BTreeMap;
...
+        }
+        for xs in data.chunks(SIMD_SIZE) {
+            let mut array = [0; SIMD_SIZE];
+            array[..xs.len()].copy_from_slice(xs);
+
+            #[cfg(not(target_feature = "avx2"))]
+            result.push(Self(array));
+
+            #[cfg(target_feature = "avx2")]
+            unsafe {
```
[the remark this answers is not in the document]
Comment (vbkaisetsu) [reply vibrato/src/dictionary/connector/raw_connector/scorer.rs:39]: fixed

File: vibrato/src/dictionary/connector/raw_connector/scorer.rs [hunk 2]
```diff
@@ -3,13 +3,112 @@ use std::collections::BTreeMap;
...
+pub struct U31Array([u32; SIMD_SIZE]);
+#[cfg(target_feature = "avx2")]
+#[derive(Clone, Copy)]
+pub struct U31Array(__m256i);
+
+impl U31Array {
+    pub fn to_simd_vec(data: &[u32]) -> Vec<Self> {
+        let mut result = vec![];
+        for &x in data {
+            assert!(x <= i32::MAX as u32);
```
[the remark this answers is not in the document]
Comment (vbkaisetsu) [reply vibrato/src/dictionary/connector/raw_connector/scorer.rs]: fixed

File: vibrato/src/num.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+const U31_VALID_RANGE: AllowedEnumVariants = AllowedEnumVariants::Range {
+    min: 0,
+    max: U31::MAX.get(),
+};
+
+impl Decode for U31 {
+    fn decode<D: Decoder>(decoder: &mut D) -> Result<Self, DecodeError> {
+        let x = Decode::decode(decoder)?;
+        if x > Self::MAX.get() {
```
Comment (kampersanda) [inline vibrato/src/num.rs:41]: I think the conditional check should employ U31::new.

**Review by kampersanda — COMMENTED** [review]
> I renamed U31Array to U31x8 and added a new struct named U31.

LGTM

**Review by kampersanda — APPROVED** [review]

**Pull request merged**